### PR TITLE
py-herbivores-and-carnivores

### PR DESCRIPTION
**Pull request by muriakula**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,29 @@
...
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
```
Comment (olena-hul) [inline app/main.py:12]: Use `()` to split lines

File: app/main.py [hunk 2]
```diff
@@ -1 +1,29 @@
...
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(food: Herbivore) -> None:
+        if isinstance(food, Herbivore):
+            if not food.hidden:
+                food.health -= 50
+                if food.health <= 0:
+                    Animal.alive.remove(food)
```
Comment (olena-hul) [inline app/main.py:29]: You can move this chek one level above to avoid lots of nesting

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1 +1,28 @@
...
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(food: Herbivore) -> None:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:24]: food - їжа
Please change variable name

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**